pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -334,14 +345,29 @@ def select_precision_plugin(self) -> PrecisionPlugin:
                             "please using native AMP for 16-bit precision."
                         )
                     log.info("Using APEX 16bit precision.")
-                    self.amp_type = AMPType.APEX
                     return ApexMixedPrecisionPlugin(self.amp_level)
         else:
             raise NotImplementedError("We only support precisions 32 and 16!")
 
-    def select_training_type_plugin(self) -> TrainingTypePlugin:
+    def _set_precision_type(self):
+        if self.precision == 32:
+            self.amp_type = None
+        elif self.precision == 16:
+            if self.amp_type == 'amp':
+                self.amp_type = AMPType.NATIVE
+            elif self.amp_type == 'apex':
+                self.amp_type = AMPType.APEX
```

Comment:
```suggestion
            self.amp_type =AMPType(self.amp_type)
```

Reply:
in fact if in case of `if self.precision == 32` also `self.amp_type is None` the you can replace all by
```
self.amp_type = AMPType.from_str(self.amp_type)
```